sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -1608,6 +1601,22 @@ class CompactBlockProcessor internal constructor(
             var downloadedBlocks = listOf<JniBlockMeta>()
             var downloadException: CompactBlockProcessorException.FailedDownloadException? = null
 
+            val fromState =
+                when (val response = downloader.getTreeState(BlockHeightUnsafe.from(batch.range.start - 1))) {
```

Comment:
This new gRPC call will be called for every batch of blocks. Is that our intention? If it is, should we move it under the retry logic below? I suppose this call can fail for the same reasons as the `downloadBlockRange` below.

Reply:
Yes, this is the intention. I didn't think that including it in the existing retry logic was correct (as we don't want to _rerun_ this gRPC call for every failed request for the batch of blocks), but I also didn't know if simple gRPC calls were being handled elsewhere or not, so I just did the simple thing to get it working. I leave it to you to decide how to handle reliability here.

Reply:
I'm adding a change to move the `TreeState` fetching into a separate function with its retry logic. Please check it and let me know if that solution best suits our needs. The LightWalletdClient logic failures are reported within the `Response` result and are not recovered anywhere else.